src/main/java/com/amazonaws/AmazonECRClientFactory.java

```diff
@@ -0,0 +1,54 @@
+package com.amazonaws;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.services.ecr.AmazonECRClient;
+import hudson.ProxyConfiguration;
+import jenkins.model.Jenkins;
+
+/**
+ * Factory to use Jenkins Proxy configuration with a AmazonECRClient.
+ *
+ * @author Marcus Wallin, kuisathaverat
+ */
+public class AmazonECRClientFactory {
+    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings(
+            value="NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE",
+            justification="The null value it is checked")
+    private final static class ClientConfigurationWrapper extends ClientConfiguration {
+        public ClientConfigurationWrapper() {
+            super();
+            Jenkins j = Jenkins.getInstance();
+            if (j != null && j.proxy != null) {
+                setProxyHost(j.proxy.name);
+                setProxyPort(j.proxy.port);
+                setProxyUsername(j.proxy.getUserName());
+                setProxyPassword(j.proxy.getPassword());
+            }
+        }
+
+    }
+
+    /**
+     * @return {@link AmazonECRClient} without proxy configuration.
+     */
+    public AmazonECRClient getAmazonECRClient() {
+        return new AmazonECRClient();
```

Comment:
I don't really get the point here. I thought the point of this factory class to return instances configured to use the Jenkins proxy configuration.